### ISSUE TITLE
UI doesn't update properly after a while being idle / not focused

### ISSUE THREAD

**Issue by nomis51** [ForkIssues/TrackerWin#1568]
# What's happening ?
The UI says "you have 10 commits to pull". I click "pull", it works, but the UI says that I still have commits that I need to pull on the branch name and in the app header (yellow rectangles). Even though the commits have been pull properly.

![image](https://user-images.githubusercontent.com/25111613/174814503-a48b0398-0d07-441e-b262-1fedd7248151.png)

Also, the UI says "no changes" next to the "Local changes" label, even though there ares changes in the files of the repo.

I noticed that it happens if Fork was on "idle" for a while (e.g. I didn't focus/interacted with the app for couple minutes). It seems to just go to "sleep" and stop inspecting the repo.

# What's expected ?
I'm expecting the app to update the UI after pulling commits or editing files in the repo, at least when I focus the app, doesn't need to do it in background to save resources. If I have 10 commits to pull, and I then I pull them, I expect the UI to say that no commit needs to be pulled afterward. If I edit files in the repo, I expect Fork to detect them, at least when I focus the app, like Git itself does, without me having to manually tell him to do so.

# Any workarounds ?
If I force a UI update, by changing repository or changing branch it updates properly, but it's quite annoying, especially if I have changes pending, so I need to stash re-apply two times in a row...

# Specs
Windows 10
Fork 1.65.2.0

**Comment by DanPristupov** [comment]
Does this happen all the time? Do you know a reliable way to reproduce that?
Do the changes appear if you press F5?

**Comment by nomis51** [comment]
No, but at least once every day.

The only way I know that causes the problem is : 
- Set Fork to be on a repo
- Alt-Tab to the IDE working on the code for that repo for a while (10-20+ mins) without touching or focusing Fork at all (or even just not having Fork focused at all for a while, not even need to work on the repo non-stop, it needs to be idle in the background)
- Come back to Fork, and now the changes / commits are not detected (not _all_ the time, but like I said at least once everyday), even if I click on Local changes, etc.
- I then need to change repo or branch to make the UI refresh

I noticed that if I somehow interact with Git for that repo (e.g. terminal, git bash, or another git client) Fork wakes up.

I didn't know I could refresh with F5, I'll try that next time it happens and tell you if it works or not. (If it does work, then it's a fine solution for me to be honest, I just need a quick non-intrusive way to tell the app to wake up when it "sleeps" :) )

**Comment by nomis51** [comment]
F5 works! Fork update the UI properly. Thanks! Didn't know about that shortcut

**Comment by conanthedev** [comment]
I was about to report the same problem. I tried to repro, but it seems intermittent. It happens at least once per day, after Fork is minimized or behind another window and I activate it by clicking on the task bar and it doesn't refresh. Glad to learn about F5!

**Comment by nomis51** [comment]
> I was about to report the same problem. I tried to repro, but it seems intermittent. It happens at least once per day, after Fork is minimized or behind another window and I activate it by clicking on the task bar and it doesn't refresh. Glad to learn about F5!

Good point, thinking about it now, it happens for me when I switch to Fork using the taskbar instead of Alt+Tab or Windows+Tab.

**Comment by DanPristupov** [comment]
@nomis51 

> Good point, thinking about it now, it happens for me when I switch to Fork using the taskbar instead of Alt+Tab or Windows+Tab.

This is a very important observation!
